terraform/modules/galileo-eks/examples/network.tf

```diff
@@ -0,0 +1,221 @@
+resource "aws_vpc" "vpc" {
+  cidr_block = "192.168.0.0/16"
+
+  enable_dns_support   = true
+  enable_dns_hostnames = true
+
+  tags = {
+    Name = "galileo_${var.environment}_vpc"
+  }
+}
+
+# Create Public Subnets
+resource "aws_subnet" "public_subnet_a" {
+  vpc_id            = aws_vpc.vpc.id
+  cidr_block        = "192.168.0.0/19"
+  availability_zone = "us-west-2a"
+  map_public_ip_on_launch = true  # Enable auto-assign public IP
+
+  tags = {
+    Name = "galileo_${var.environment}_public_subnet_a"
+  }
+}
+
+resource "aws_subnet" "public_subnet_b" {
+  vpc_id            = aws_vpc.vpc.id
+  cidr_block        = "192.168.32.0/19"
+  availability_zone = "us-west-2b"
+  map_public_ip_on_launch = true
+
+  tags = {
+    Name = "galileo_${var.environment}_public_subnet_b"
+  }
+}
+
+resource "aws_subnet" "public_subnet_c" {
+  vpc_id            = aws_vpc.vpc.id
+  cidr_block        = "192.168.64.0/19"
+  availability_zone = "us-west-2c"
+  map_public_ip_on_launch = true
+
+  tags = {
+    Name = "galileo_${var.environment}_public_subnet_c"
+  }
+}
+
+# Create Private Subnets
+resource "aws_subnet" "private_subnet_a" {
+  vpc_id            = aws_vpc.vpc.id
+  cidr_block        = "192.168.96.0/19"
+  availability_zone = "us-west-2a"
+
+  tags = {
+    Name = "galileo_${var.environment}_private_subnet_a"
+  }
+}
+
+resource "aws_subnet" "private_subnet_b" {
+  vpc_id            = aws_vpc.vpc.id
+  cidr_block        = "192.168.128.0/19"
+  availability_zone = "us-west-2b"
+
+  tags = {
+    Name = "galileo_${var.environment}_private_subnet_b"
+  }
+}
+
+resource "aws_subnet" "private_subnet_c" {
+  vpc_id            = aws_vpc.vpc.id
+  cidr_block        = "192.168.160.0/19"
+  availability_zone = "us-west-2c"
+
+  tags = {
+    Name = "galileo_${var.environment}_private_subnet_c"
+  }
+}
+
+# Internet Gateway
+resource "aws_internet_gateway" "igw" {
+  vpc_id = aws_vpc.vpc.id
+
+  tags = {
+    Name = "galileo_${var.environment}_igw"
+  }
+}
+
+# Elastic IP for NAT Gateway in Subnet A
+resource "aws_eip" "nat_eip_a" {
+  domain = "vpc"
+}
```

Comment:
Have we checked if an IPv6 EIP would work? Asking because IPv4s [will start to get charged](https://aws.amazon.com/blogs/aws/new-aws-public-ipv4-address-charge-public-ip-insights/) ~soon.

Reply:
I'm not sure, I know that the VPC setup by `eksctl` didn't use them so I didn't test this functionality.